### ISSUE TITLE
Can I train PPO for more than 10 Million steps?

### ISSUE THREAD

**Issue by knmitri** [Unity-Technologies/ml-agents#4193]
I am training a complicated model that needs lot of training steps to converge. For several runs (with different settings and environments), my training loss is suddenly increasing drastically after 10 million steps as shown in the picture. Before I did not have this issue, is there a limit in the ML agent release 3 for training steps?

![image](https://user-images.githubusercontent.com/30293888/86458381-71dd7580-bd25-11ea-823c-96f5859925c8.png)

- Unity Version: [e.g. Unity 2019.3.6f1]
- OS + version: [e.g. Windows 10]
- _ML-Agents version_: (e.g. ML-Agents v3.0)
- _TensorFlow version_: 2.2.0
- _Environment_: custom environment

**Comment by Shubhamai** [comment]
Do you have enabled curiosity? What was the cumulative reward graph?

**Comment by knmitri** [comment]
> Do you have enabled curiosity? What was the cumulative reward graph?
@Shubhamai 
No I did not enable curiosity. the cumulative reward is also decreasing to large negative values.

**Comment by Shubhamai** [comment]
@knmitri  I don't think there is a problem or bug somewhere, i would say that is normal

**Comment by zaiyun** [comment]
I had the exact same issue on my model. I don't have this problem when using another pc with an older Mlagent version. With the exact same project and same setting and everything, the newer version mlagent will collapse after certain steps. This is definitely a bug. 
![1](https://user-images.githubusercontent.com/47151605/89118900-a30daa80-d477-11ea-9c12-a8049b273b00.PNG)

**Comment by niskander** [comment]
@knmitri  @zaiyun just curious what versions were you upgrading from (that did not have this issue)?

I've seen this happen with SAC (training is going well, no errors in training environment, then all of a sudden they stop playing), and this happened whenever I used negative rewards, which @zaiyun also seems to use. 

However I don't have this issue with PPO, at least not in version 1.0.2. I can train up to 40M+ steps.

**Comment by zaiyun** [comment]
I did not upgrade it from an older version. I directly installed the newest version. And on another pc, I installed the older version.

**Comment by vincentpierre** [comment]
Hi,
Is there a way to reproduce this issue with one of the example environments on the latest version of ML-Agents?
I am unable to reproduce this issue.

**Comment by hvpeteet** [comment]
Closing since stale
